app/src/main/java/io/capibaras/abcall/viewmodels/NavigationViewModel.kt

```
@@ -0,0 +1,86 @@
+
+    private fun checkLogoutStatus() {
+        if (!isUserLoggedIn && isManualLogout) {
+            successUIState = SuccessUIState.Success(R.string.success_logout)
+        }
+    }
+
+    private fun checkUserSession() {
+        try {
+            val token = tokenManager.getAuthToken()
+            isUserLoggedIn = !token.isNullOrEmpty()
+        } catch (e: Exception) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)